buffer/src/main/java/io/netty/buffer/ByteBufUtil.java

```diff
@@ -464,6 +464,7 @@ public static int writeUtf8(ByteBuf buf, CharSequence seq) {
 
     // Fast-Path implementation
     static int writeUtf8(AbstractByteBuf buffer, int writerIndex, CharSequence seq, int len) {
+        buffer.ensureAccessible();
```

Comment:
it seems more natural/consistent for the ByteBuf implementation to enforce this "internal" constraint instead of enforcing this externally. However I guess this is the equivalent of the "set" operations in the ByteBuf implementations which typically provide the accessible check ... so maybe this is OK.